ssz_test.go

```diff
@@ -0,0 +1,237 @@
+package ssz
```

Comment:
No review needed here, simply consolidated the tests of `unmarshal_test.go, marshal_test.go, hash_tree_root_test.go, signing_root_test.go` into a single file